### PR TITLE
Milestone 8 : Integration specs for views and fix n+1 problems (Pair-programming)

### PR DESCRIPTION
**Pull request by ABDULALI3468**
> ## What we did ♻️
> In this project's milestone, we create integration tests for all of the views used in this project.
> 
> ## Project requirements 📗
> #### n+1
> * Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.
> 
> #### Integration specs
> * Use Capybara to write integration tests for each view in your project.
> * User index page:
>   
>   * I can see the username of all other users.
>   * I can see the profile picture for each user.
>   * I can see the number of posts each user has written.
>   * When I click on a user, I am redirected to that user's show page.
> * user show page:
>   
>   * I can see the user's profile picture.
>   * I can see the user's username.
>   * I can see the number of posts the user has written.
>   * I can see the user's bio.
>   * I can see the user's first 3 posts.
>   * I can see a button that lets me view all of a user's posts.
>   * When I click a user's post, it redirects me to that post's show page.
>   * When I click to see all posts, it redirects me to the user's post's index page.
> * User post index page:
>   
>   * I can see the user's profile picture.
>   * I can see the user's username.
>   * I can see the number of posts the user has written.
>   * I can see a post's title.
>   * I can see some of the post's body.
>   * I can see the first comments on a post.
>   * I can see how many comments a post has.
>   * I can see how many likes a post has.
>   * I can see a section for pagination if there are more posts than fit on the view.
>   * When I click on a post, it redirects me to that post's show page.
> * Post show page:
>   
>   * I can see the post's title.
>   * I can see who wrote the post.
>   * I can see how many comments it has.
>   * I can see how many likes it has.
>   * I can see the post body.
>   * I can see the username of each commentor.
>   * I can see the comment each commentor left.
> 
> **Be sure to include integration specs for any other features you may have implemented!**

### CODE REVIEW

File: spec/features/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
+require 'rails_helper'
+
+RSpec.feature 'Posts/index', type: :feature do
```
Comment (SajjadAhmad14) [inline spec/features/posts_index_spec.rb:4]: Please add a test that asserts you can see a section for pagination if there are more posts than fit on the view on the `User post index` page.

File: spec/features/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.feature 'Post/show', type: :feature do
```
Comment (SajjadAhmad14) [inline spec/features/posts_show_spec.rb:3]: Please add a test that asserts you can see the username of each commentor on the `User post show` page.

Please add a test that asserts you can see the comment each commentor left on the `User post show` page.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Abdul, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏
- [ ] Tests are failing due to some reasons. Kindly see the screenshots below and make sure all the tests are green.


test1

test2

test3


tests4

test5


test8



- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/features/posts_show_spec.rb (hunk 2 above)
Comment (ABDULALI3468) [reply spec/features/posts_show_spec.rb:3]: @SajjadAhmad14 With deep respect! I think you have not run the test correctly. Remember you also need to convert config/database.yml to change the password according to your database and also run the command rails db:reset
Screenshot 2022-09-28 211548

**Comment by ABDULALI3468** [comment]
@SajjadAhmad14 Moreover` this project is related to feature section` only. So i will appreciate if you run the following command only `rspec spec/features`

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey TEAM! 😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ABDULALI3468** [comment]
@zainsadaqat With respect thank you so much for approving my project.

**Pull request merged**